pkg/common/secret.go

```diff
@@ -282,3 +286,55 @@ func DeleteSecretsWithLabel(r ReconcilerCommon, obj metav1.Object, labelSelector
 
 	return nil
 }
+
+//
+// GetDataFromSecret - Get data from Secret
+//
+// if the secret or data is not found, requeue after requeueTimeout in seconds
+func GetDataFromSecret(
+	r ReconcilerCommon,
+	object client.Object,
+	cond *ospdirectorv1beta1.Condition,
+	conditionDetails ospdirectorv1beta1.ConditionDetails,
+	secretName string,
+	requeueTimeout int,
+	key string,
+) (string, ctrl.Result, error) {
+
+	data := ""
+
+	secret, _, err := GetSecret(r, secretName, object.GetNamespace())
+	if err != nil {
+		if k8s_errors.IsNotFound(err) {
+			cond.Message = fmt.Sprintf("%s secret does not exist: %v", secretName, err)
+			cond.Reason = ospdirectorv1beta1.ConditionReason(conditionDetails.ConditionNotFoundReason)
+			cond.Type = ospdirectorv1beta1.ConditionType(conditionDetails.ConditionNotFoundType)
+
+			LogForObject(r, cond.Message, object)
+
+			return data, ctrl.Result{RequeueAfter: time.Duration(requeueTimeout) * time.Second}, nil
+		}
+		cond.Message = fmt.Sprintf("Error getting %s Secret: %v", secretName, err)
+		cond.Reason = ospdirectorv1beta1.ConditionReason(conditionDetails.ConditionErrordReason)
+		cond.Type = ospdirectorv1beta1.ConditionType(conditionDetails.ConditionErrorType)
+		err = WrapErrorForObject(cond.Message, object, err)
+
+		return data, ctrl.Result{}, err
+	}
+
+	if key != "" {
+		if _, ok := secret.Data[key]; !ok {
```

Comment:
just a minor nit, but you could as well save the key for an externally (to if) scoped variable for future use at the line #336 and save an extra Data[key] accessing off

Reply:
yes correct, let me change that